### DISCUSSION TITLE
set contraints on step width

### DISCUSSION THREAD

**Discussion by linghuix** [KULeuvenNeuromechanics/PredSim#175] [Q&A]
Hi, I want to know how to fix step width in the simulation. I couldn't find a parameter related to it in the S object. Is there a way to do it?

**Comment by ytlin2021** [comment]
Hi, I believe you can use S.bounds.distanceConstraints to enforce a minimum or maximum distance between two points. For example, to constrain the width between two foot contact points, you could do something like this:
```
% define points 
S.bounds.points(end+1).body = 'Foot_L'; 
S.bounds.points(end).point_in_body = [ ];  % Replace with the point in your model
S.bounds.points(end).name = 'FootContact1_l'; 

S.bounds.points(end+1).body = 'Foot_R'; 
S.bounds.points(end).point_in_body = [ ]; % Replace with the point in your model
S.bounds.points(end).name = 'FootContact1_r';

% constrain minimal distance between 2 points 
S.bounds.distanceConstraints(end+1).point1 = 'FootContact1_r'; 
S.bounds.distanceConstraints(end).point2 = 'FootContact1_l'; 
S.bounds.distanceConstraints(end).direction = 'xz'; 
S.bounds.distanceConstraints(end).lower_bound = [ ]; % Replace with the minimum width between feet
S.bounds.distanceConstraints(end).upper_bound = [ ]; % Replace with the maximum width between feet
```

This imposes a distance constraint along the 'xz' plane between the foot contact points. However, I'm unsure how to strictly fix the step width. Any other suggestions or insights are welcome!

**Reply by linghuix** [reply]
Thanks for your answer. It works. I am now trying this code to constrain step width:
```
   % set constraints on step width  
    S.bounds.distanceConstraints(1).point1 = 'calcn_r';
    S.bounds.distanceConstraints(1).point2 = 'calcn_l';
    S.bounds.distanceConstraints(1).direction = 'z';
    S.bounds.distanceConstraints(1).lower_bound = 0.06;
    S.bounds.distanceConstraints(1).upper_bound = 0.07;
```
But this constrains the distance of two points along z axis during the whole gait period. I am not sure this is right because the step width is typically defined as the distance along z axis between heel strikes of left and right leg. This adds extra constraints during swing phase, it may cause some differences in the gait pattern. If there is any other insights about this, welcome to discuss.
